va/caa_test.go

```diff
@@ -578,13 +575,400 @@ func TestIsCAAValidParams(t *testing.T) {
+			expectedProbType:      probs.DNSProblem,
+			remoteVAs: []RemoteVA{
+				{remoteVA, remoteUA},
+				{remoteVA, remoteUA},
+				{remoteVA, remoteUA},
+			},
+		},
+		{
+			name:                     "functional localVA, 1 broken RVA, no CAA records",
+			domains:                  "present-dns-only.com",
+			expectedProbSubstring:    "During secondary CAA checking: While processing CAA",
+			expectedProbType:         probs.DNSProblem,
+			expectedDiffLogSubstring: `RemoteSuccesses":2,"RemoteFailures":[{"VAHostname":"broken","Problem":{"type":"dns","detail":"While processing CAA`,
+			localDNSClient:           caaMockDNS{},
+			remoteVAs: []RemoteVA{
+				{brokenVA, brokenUA},
+				{remoteVA, remoteUA},
+				{remoteVA, remoteUA},
+			},
+		},
+		{
+			name:                     "functional localVA, all broken RVAs, no CAA records",
+			domains:                  "present-dns-only.com",
+			expectedProbSubstring:    "During secondary CAA checking: While processing CAA",
+			expectedProbType:         probs.DNSProblem,
+			expectedDiffLogSubstring: `RemoteSuccesses":0,"RemoteFailures":[{"VAHostname":"broken","Problem":{"type":"dns","detail":"While processing CAA for`,
+			localDNSClient:           caaMockDNS{},
+			remoteVAs: []RemoteVA{
+				{brokenVA, brokenUA},
+				{brokenVA, brokenUA},
+				{brokenVA, brokenUA},
+			},
+		},
+		{
+			name:           "all VAs functional, CAA issue type present",
+			domains:        "present.com",
+			localDNSClient: caaMockDNS{},
+			remoteVAs: []RemoteVA{
+				{remoteVA, remoteUA},
+				{remoteVA, remoteUA},
+				{remoteVA, remoteUA},
+			},
+		},
+		{
+			name:                     "functional localVA, 1 broken RVA, CAA issue type present",
+			domains:                  "present.com",
+			expectedProbSubstring:    "During secondary CAA checking: While processing CAA",
+			expectedProbType:         probs.DNSProblem,
+			expectedDiffLogSubstring: `RemoteSuccesses":2,"RemoteFailures":[{"VAHostname":"broken","Problem":{"type":"dns","detail":"While processing CAA for`,
+			localDNSClient:           caaMockDNS{},
+			remoteVAs: []RemoteVA{
+				{brokenVA, brokenUA},
+				{remoteVA, remoteUA},
+				{remoteVA, remoteUA},
+			},
+		},
+		{
+			name:                     "functional localVA, all broken RVAs, CAA issue type present",
+			domains:                  "present.com",
+			expectedProbSubstring:    "During secondary CAA checking: While processing CAA",
+			expectedProbType:         probs.DNSProblem,
+			expectedDiffLogSubstring: `RemoteSuccesses":0,"RemoteFailures":[{"VAHostname":"broken","Problem":{"type":"dns","detail":"While processing CAA for`,
+			localDNSClient:           caaMockDNS{},
+			remoteVAs: []RemoteVA{
+				{brokenVA, brokenUA},
+				{brokenVA, brokenUA},
+				{brokenVA, brokenUA},
+			},
+		},
+		{
+			// The RVA results are dropped because the localVA lookup determined
+			// issuance was forbidden and never fired off the gRPCs to the RVAs
+			// meaning that no differential was created.
```

Comment:
The localVA kicks off the background goroutines before doing its own check. But if its own check fails, it doesn't wait for their results.